### PR TITLE
Add flags on batch command.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
## Fixes / Features
- Add flags on batch command.

## Testing / Documentation
Testing details.

- [ y/n ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Review by IrvingMg — DISMISSED** [review]
LGTM.

File: src/xpk/commands/info.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def get_flavors_resources_reservations(
   reservations[cq_name] = {}
   for flavor_name, flavor_resources_reservation_list in flavors_res.items():
     for resource in flavor_resources_reservation_list:
-      reservations[cq_name][f'{flavor_name}:{resource["name"]}'] = resource[
+      reservations[cq_name][f'{flavor_name}:{resource.name}'] = resource[
```
Comment (pawloch00) [inline src/xpk/commands/info.py:165]: This is not connected with adding batch flags.  As it is minor change it may be here, but please do not unrelated changes to PR

Comment (mbobrovskyi) [reply src/xpk/commands/info.py:165]: Just fix for linter.

File: src/xpk/commands/workload.py [hunk 2]
```diff
@@ -374,7 +374,7 @@ def workload_create(args) -> None:
   ):
     xpk_print(
         'Warning: Cluster has been created using XPK version:'
-        f' {cluster_config_map["xpk_version"]} but the XPK version you are'
+        f' {cluster_config_map.xpk_version} but the XPK version you are'
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/workload.py:377]: Yeah, the same situation. Just to fix linter. I can also move it to separate PR.

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/workload.py:377]: However, in the main branch, we don't have this issue; it only occurs after these changes.

File: src/xpk/commands/batch.py [hunk 3]
```diff
@@ -46,11 +48,66 @@ def batch(args: Namespace) -> None:
...
+  if args.mem_per_task is not None:
+    cmd += f' --mem-per-task {args.mem_per_task}'
+
+  if args.mem_per_cpu is not None:
+    cmd += f' --mem-per-cpu {args.mem_per_cpu}'
+
+  if args.mem_per_gpu is not None:
+    cmd += f' --mem-per-gpu {args.mem_per_gpu}'
+
+  if args.nodes is not None:
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py:84]: On kjob we don't have constraints for this flag.

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py:84]: What max number of nodes we should set? WDYT?

File: src/xpk/parser/batch.py [hunk 4]
```diff
@@ -39,3 +41,157 @@ def set_batch_parser(batch_parser):
...
+      action=argparse.BooleanOptionalAction,
+      default=False,
+      help="Enable the retrieval of the first node's IP address.",
+  )
+  # TODO: Find solution for time.Duration()
+  # batch_optional_arguments.add_argument(
+  #   '--first-node-ip-timeout',
+  #   type=int,
+  #   default=False,
+  #   help='The timeout for the retrieval of the first node\'s IP address.'
```
Comment (mbobrovskyi) [inline src/xpk/parser/batch.py:64]: I also can't find Duration type for this flag. Maybe we should set as string and add custom validation. 

@IrvingMg @pawloch00 WDYT?

File: src/xpk/commands/batch.py [hunk 5]
```diff
@@ -46,11 +48,66 @@ def batch(args: Namespace) -> None:
...
-      f' {APP_PROFILE_TEMPLATE_DEFAULT_NAME} --'
-      f' {args.script}'
+      'kubectl kjob create slurm'
+      f' --profile {APP_PROFILE_TEMPLATE_DEFAULT_NAME}'
+      f' --localqueue {LOCAL_QUEUE_NAME}'
   )
 
+  if args.ignore_unknown_flags:
+    cmd += ' --ignore-unknown-flags'
+  if args.first_node_ip:
```
[the remark this answers is not in the document]
Comment (pawloch00) [reply src/xpk/commands/batch.py:58]: want this interfere with xpk flags? I eam first_node_ip :)

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py:58]: Ah, sorry :)

[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py]: Removed

File: src/xpk/parser/batch.py (hunk 4 above)
Comment (mbobrovskyi) [reply src/xpk/parser/batch.py]: Removed

File: src/xpk/commands/batch.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py:82]: Any thoughts?

[the remark this answers is not in the document]
Comment (IrvingMg) [reply src/xpk/commands/batch.py:82]: Based on the command used to create clusters on GKE, I'd say that we can set 1000 as the max limit:

https://github.com/AI-Hypercomputer/xpk/blob/c1551023d58599e3cbae76e559ebae6a23c7aa82/src/xpk/commands/cluster.py#L479

File: src/xpk/parser/batch.py [hunk 6]
```diff
@@ -39,3 +41,137 @@ def set_batch_parser(batch_parser):
...
+      '--mem-per-gpu',
+      type=str,
+      default=None,
+      help=(
+          'How much memory a container requires, it multiplies the number '
+          'of requested gpus per task by mem-per-gpu.'
+      ),
+  )
+  batch_optional_arguments.add_argument(
+      '-N',
```
Comment (pawloch00) [inline src/xpk/parser/batch.py:112]: -N for n. of nodes and -n for n of tasks is bad idea for me

**Review by pawloch00 — APPROVED** [review]

**Comment by pawloch00** [comment]
@44past4 please take a look

File: src/xpk/parser/batch.py (hunk 6 above)
Comment (mbobrovskyi) [reply src/xpk/parser/batch.py:112]: Slurm uses these flags https://slurm.schedmd.com/sbatch.html#OPT_nodes.

**Review by 44past4 — APPROVED** [review]

File: src/xpk/commands/batch.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply src/xpk/commands/batch.py:82]: Ah. Not sure. This is nodes. But nodes on the Slurm it is pods.

[the remark this answers is not in the document]
Comment (IrvingMg) [reply src/xpk/commands/batch.py:82]: Then, probably we can use 110? It seems that's the default max num of pods per node on a GKE cluster. Documentation: https://cloud.google.com/kubernetes-engine/docs/how-to/flexible-pod-cidr#max_pods_default

Looks like it's possible to change that value, but I don´t see we're changing it in xpk.

[the remark this answers is not in the document]
Comment (IrvingMg) [reply src/xpk/commands/batch.py:82]: Or based on the info here https://kubernetes.io/docs/setup/best-practices/cluster-large/, should we set a limit of 150K? @pawloch00 WDYT?

[the remark this answers is not in the document]
Comment (pawloch00) [reply src/xpk/commands/batch.py:82]: Ok, let's leave it unconstrained, If I understand https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#job-v1-batch correctly, it will be checked.

**Review by pawloch00 — APPROVED** [review]

**Review by IrvingMg — DISMISSED** [review]

**Review by IrvingMg — APPROVED** [review]

**Review by pawloch00 — APPROVED** [review]
